### PR TITLE
adding a "retryable fail" event for a test that fails but can be retried

### PR DESCRIPTION
**Pull request by catdad**
Addresses issue #2529 by allowing reporters to take action on tests that have failed but can be retried.

### CODE REVIEW

**Comment by ScottFreeCode** [comment]
Thanks for the submission!

As I mentioned on #2529, I don't have strong opinions for or against the feature; but I do want to put in here that I like that the implementation is this simple, and especially that you've provided a test (and at a glance it looks like that function might not have been tested before to begin with, too, so if there are objections to the feature we might still want a similar test...). :+1: Nicely done!

**Comment by catdad** [comment]
@ScottFreeCode, thanks for the feedback. If there is anything I can do to make this go faster or smoother, please let me know.

**Comment by LToronto** [comment]
👍  well done @catdad this fill my need very nicely.

**Comment by mamanakisdm** [comment]
This is a GREAT idea!
We had this need so bad that one of our QA spent some time and developed a similar solution, and we thought about PUSHING the idea into this Repo. But then, he found this solution. 
Please, for the love of all that is holy, MERGE this Pull Request!

**Comment by catdad** [comment]
I updated this PR with the latest code from master, since there were conflicts. Though, it's been 10 months now, so I am starting to lose hope.

File: test/unit/runner.spec.js [hunk 1]
```diff
@@ -385,6 +385,40 @@ describe('Runner', function() {
...
+      var ERR = new Error('bear error');
+
+      var TEST = new Test('im a test about bears', function() {
+        if (retryableFails < retries) {
+          throw ERR;
+        }
+      });
+
+      TEST.retries(retries);
+      suite.retries(retries);
```
Comment (outsideris) [inline test/unit/runner.spec.js]: `suite.retries(retries);` is enought for tests. 
IMO `TEST.retries(retries);` not unneccesary here.

Comment (catdad) [reply test/unit/runner.spec.js]: Thanks for the feedback. I agree and have updated the PR.

**Comment by outsideris** [comment]
Markdown linting is weired. Current master branch has been fixed.
It it rebased on master, CI will pass.

**Comment by boneskull** [comment]
I'll see if I can merge this manually.

**Comment by boneskull** [comment]
see #3408

**Pull request closed without merging**